johnbywater/eventsourcing
# Improve contributor expirience

Hi @johnbywater!

Recently, I wanted to contribute independent Django infrastructure tables per application (see this conversation https://github.com/johnbywater/eventsourcing/issues/199#issuecomment-599138297) but it was a bit unclear for me how to start the contribution process from my side. Mainly because I didn't find guides about the process: 
- what the default description of a pull request
- should be there an open issue for it and how it should look alike
- project code rules how it should be formatted
- how to run tests locally

So, this is my very draft proposal with some changes to the local development workflow. I'm open to the discussion your feel about these changes.

The full list of the proposed changes (in no order about importance):
- Add `.dockerignore` file: used to speed up the image building by reducing the number of files passed to the Docker daemon for the image build.
- Add `.env` file and updated `dev/docker-compose.yaml` to have a cleaner way to load this environment and used them with `docker-compose` utility. Also, I opened service ports to ease debugging.
- Updated `Contributing` documentation.
- Updated `.editorconfig` file for YAML files.
- Add `isort`, `flake8` and `flake8-bugbear` packages to lint the code
- Add lint checks into TravisCI configuration file
- Add the `Makefile` approach at the root folder of the project
- Reformatted the code with isort and black (it went wild without lint check on CI)
- Add .git-blame-ignore-revs to preserve repository history

Anyway, I wasn't able to run the tests successfully at the local machine. So my next steps about it:
- find out how to do it locally (asking you about it, of course)
- update some documentation again
- fixup all lint errors that presented right now (flake and mypy for sure)

Best!
